src/python/pants/backend/shell/goals/test.py

```diff
@@ -88,6 +97,31 @@ async def test_shell_command(
     )
 
 
+@rule(desc="Test with shell command (interactively)", level=LogLevel.DEBUG)
+async def test_shell_command_interactively(
+    batch: ShellTestRequest.Batch[TestShellCommandFieldSet, Any],
+) -> TestDebugRequest:
+    field_set = batch.single_element
+    wrapped_tgt = await Get(
+        WrappedTarget,
+        WrappedTargetRequest(field_set.address, description_of_origin="<infallible>"),
+    )
+
+    shell_process = await Get(
+        Process,
+        ShellCommandProcessFromTargetRequest(wrapped_tgt.target),
+    )
+
+    # This is probably not strictly necessary given the use of `InteractiveProcess` but good to be correct in any event.
+    shell_process = dataclasses.replace(shell_process, cache_scope=ProcessCacheScope.PER_SESSION)
+
+    return TestDebugRequest(
+        InteractiveProcess.from_process(
+            shell_process, forward_signals_to_process=False, restartable=True
```

Comment:
More of a question about `pants test --debug` in general: Should `restartable` be configurable somehow?

Reply:
Good question. It's occasionally annoying if I forget that `--debug` restarts on file change, and so I make a few edits, save, and find my in-progress debugging session interrupted. 

I don't know if I'd remember to pass an extra flag like `pants test --debug --no-debug-restart` or whatever.

Reply:
Maybe such a flag should be specific to the kind of test then (and then configured in `pants.toml`)?

Reply:
In any event, going to land this PR since I'm just matching existing behavior of our `test --debug` impls here.